warehouse/core/src/main/java/datawave/iterators/filter/ConfigurableAgeOffFilter.java

```diff
@@ -336,46 +336,39 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
      */
     private void initFilterRules() throws IllegalArgumentException, IOException {
         // filename
-        if (null == ruleCache) {
-            synchronized (ConfigurableAgeOffFilter.class) {
-                if (null == ruleCache) {
-                    UPDATE_INTERVAL_MS = getLongProperty(UPDATE_INTERVAL_MS_PROP, DEFAULT_UPDATE_INTERVAL_MS); // 5 ms
-                    EXPIRATION_INTERVAL_MS = getLongProperty(EXPIRATION_INTERVAL_MS_PROP, DEFAULT_EXPIRATION_INTERVAL_MS); // 1 hour
-                    ruleCache = CacheBuilder.newBuilder().refreshAfterWrite(UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS)
-                                    .expireAfterAccess(EXPIRATION_INTERVAL_MS, TimeUnit.MILLISECONDS).build(new ReloadableCacheBuilder());
-                    // this will schedule a check to see if the update or expiration intervals have changed
-                    // if so the ruleCache will be rebuilt with these new intervals
-                    SimpleTimer.getInstance(1).schedule(
-                                    () -> {
-                                        try {
-                                            long interval = getLongProperty(UPDATE_INTERVAL_MS_PROP, DEFAULT_UPDATE_INTERVAL_MS);
-                                            long expiration = getLongProperty(EXPIRATION_INTERVAL_MS_PROP, DEFAULT_EXPIRATION_INTERVAL_MS);
-                                            if (UPDATE_INTERVAL_MS != interval || EXPIRATION_INTERVAL_MS != expiration) {
-                                                log.info("Changing " + UPDATE_INTERVAL_MS_PROP + " to " + interval);
-                                                UPDATE_INTERVAL_MS = interval;
-                                                log.info("Changing " + EXPIRATION_INTERVAL_MS_PROP + " to " + expiration);
-                                                EXPIRATION_INTERVAL_MS = expiration;
-                                                ruleCache = CacheBuilder.newBuilder().refreshAfterWrite(UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS)
-                                                                .expireAfterAccess(EXPIRATION_INTERVAL_MS, TimeUnit.MILLISECONDS)
-                                                                .build(new ReloadableCacheBuilder());
-                                            }
-                                        } catch (Throwable t) {
-                                            log.error(t, t);
+        synchronized (ConfigurableAgeOffFilter.class) {
+            if (null == ruleCache) {
```

Comment:
I thought we liked keeping one null check outside of the synchronized block to eliminate majority of the lock contention. As it stands now, we contend the lock on every invocation. With a check both inside and outside, we only contend the lock if the instance isn't set.

Reply:
I agree with @jschmidt10 's comment here and I do not see a response.  Synchronization can be expensive and it would be could to avoid it if possible.  I think the check outside and inside the synchronization block is desired here.

Reply:
Not sure how I missed this message.  Been trying to get sonarqube up today - current issue is the docker image pull limit.  Hopefully tomorrow I can run this early and get a scan going.
I can put the first null check back - unsure if the volatile will make a difference, but we can just ignore its guidance and do the double null check.

Reply:
OK - making it volatile doesn't help -- https://wiki.sei.cmu.edu/confluence/display/java/CON50-J.+Do+not+assume+that+declaring+a+reference+volatile+guarantees+safe+publication+of+the+members+of+the+referenced+object
I'll leave the double-check in place.